### PR TITLE
feat(integrations): Add Anthropic Integration

### PR DESCRIPTION
**Pull request by czyber**
This PR adds an anthropic integration. It supports the creation of messages in streaming and non-streaming mode.
It is heavily inspired/analogous to the OpenAI integration. 

Docs PR: https://github.com/getsentry/sentry-docs/pull/9469

---

## General Notes

Thank you for contributing to `sentry-python`!

Please add tests to validate your changes, and lint your code using `tox -e linters`.

Running the test suite on your PR might require maintainer approval. Some tests (AWS Lambda) additionally require a maintainer to add a special label to run and will fail if the label is not present.

#### For maintainers

Sensitive test suites require maintainer review to ensure that tests do not compromise our secrets. This review must be repeated after any code revisions.

Before running sensitive test suites, please carefully check the PR. Then, apply the `Trigger: tests using secrets` label. The label will be removed after any code changes to enforce our policy requiring maintainers to review all code revisions before running sensitive tests.

### CODE REVIEW

**Comment by antonpirker** [comment]
Hey @czyber !

Thanks for this great PR! 
As we released a big update to our SDK today, some of your code is broken!
I will allow myself to update your code to work with the new 2.0 version of the SDK.

**Comment by antonpirker** [comment]
Hey again @czyber , 

I made some changes to better deal with the spans. Everything looks great!

One question: Do the test actual send data to Anthropic? Or does the "z" api_key enable some sort of test mode? (because we do not want to access the internet in our tests, if it is not absolutely necessary)

**Comment by czyber** [comment]
Hey @antonpirker,

Thank you for updating the PR! 

The test do not send data to Anthropic, yet initializing the client with `api_key="z"` does not enable some kind of test mode, it does not send a request. Also, i mocked the part where requests would actually be sent (the `_post` attribute of `SyncAPIResource`)

**Review by antonpirker — APPROVED** [review]
This looks good! Thanks again for the contribution!

I will hold on merging this for a bit, because we released 2.0 yesterday and maybe we need to do some hot fixes. 

We also have a Langchain integration in the works right now, and I would like to release both in the same version, so we have a kind of AI update :-)

File: sentry_sdk/integrations/anthropic.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            output_tokens = usage.output_tokens
+
+    total_tokens = input_tokens + output_tokens
+
+    if total_tokens != 0:
+        span.set_data(TOTAL_TOKENS_USED, total_tokens)
+    if input_tokens != 0:
+        span.set_data(PROMPT_TOKENS_USED, input_tokens)
+    if output_tokens != 0:
+        span.set_data(COMPLETION_TOKENS_USED, output_tokens)
```
Comment (colin-sentry) [inline sentry_sdk/integrations/anthropic.py:78]: Once langchain is merged, this should use the new helper function https://github.com/getsentry/sentry-python/pull/2911/files#diff-c566959db6701d8426b02bcf30caa0222d01aa9931ce9913b1a35fedfd0953c1R59

File: sentry_sdk/integrations/anthropic.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        except Exception as exc:
+            _capture_exception(exc)
+            span.__exit__(None, None, None)
+            raise exc from None
+
+        integration = sentry_sdk.get_client().get_integration(AnthropicIntegration)
+
+        with capture_internal_exceptions():
+            span.set_data("ai.model_id", model)
+            span.set_data("ai.streaming", False)
```
Comment (colin-sentry) [inline sentry_sdk/integrations/anthropic.py:114]: It'd be nice to pull the tags we have in langchain too if any exist in the params: https://github.com/getsentry/sentry-python/pull/2911/files#diff-bba846cf7b623cb617bd628d482cd71fe41160f0cea1f48f321fcf5166472925R52-R62

File: sentry_sdk/integrations/anthropic.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+                                    continue
+                            yield event
+
+                        if should_send_default_pii() and integration.include_prompts:
+                            complete_message = "".join(content_blocks)
+                            span.set_data(
+                                "ai.responses",
+                                [{"type": "text", "text": complete_message}],
+                            )
+                        span.set_data(TOTAL_TOKENS_USED, input_tokens + output_tokens)
```
Comment (colin-sentry) [inline sentry_sdk/integrations/anthropic.py:166]: New helper function again (after langchain merged)

File: tests/integrations/anthropic/test_anthropic.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    if send_default_pii and include_prompts:
+        assert span["data"]["ai.input_messages"] == messages
+        assert span["data"]["ai.responses"] == [
+            {"type": "text", "text": "Hi, I'm Claude."}
+        ]
+    else:
+        assert "ai.input_messages" not in span["data"]
+        assert "ai.responses" not in span["data"]
+
+    assert span["data"][PROMPT_TOKENS_USED] == 10
```
Comment (colin-sentry) [inline tests/integrations/anthropic/test_anthropic.py:93]: Some of this has changed a bit in the langchain merge as well, it's span["measurements"] I believe

**Review by colin-sentry — CHANGES_REQUESTED** [review]
A few changes needed after langchain is merged, but this looks good otherwise. Nice work :)

**Comment by czyber** [comment]
Hey @colin-sentry ,
thanks for the review!

I have the langchain PR on my mind, and will incorporate the requested changes as soon as it is merged 👍

**Comment by antonpirker** [comment]
fyi @czyber the langchain branch has been merged into master!

**Comment by czyber** [comment]
fyi @colin-sentry @antonpirker , I updated the PR to use the shared functionality for AI integrations that was introduced in the langchain integration PR.

**Review by colin-sentry — APPROVED** [review]
Looks good to me! Nice work.

**Comment by antonpirker** [comment]
Yes! Great work everybody! Thanks a lot!

**Pull request merged**